### PR TITLE
helm_template: add optional show_only and release_namespace arguments

### PR DESCRIPTION
**Pull request by cvstebut**
##### SUMMARY

This PR adds the "show_only" and "release_namespace" as optional arguments to the helm_template module.
It does some work towards #313.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

changelogs/fragments/313-helm-template-add-support-for-show-only-and-release-namespace.yml
plugins/modules/helm_template.py
tests/unit/modules/test_helm_template.py

##### ADDITIONAL INFORMATION

The PR does include unit tests instead of integration test.

Reasoning: 

The existing integration tests already include a task based on helm_template. So we know that the module does a proper job of using the command line generated inside the module to call helm.
As I trust helm itself to "do its job" correctly, all that should be necessary is to test the correct generation of the command line itself. The included unit tests hopefully do a proper job.

With regards of the pretty long testing times for the module, I really prefer unit tests, if at all possible.

Please let me know if this fits.

```paste below

```

### CODE REVIEW

**Comment by cvstebut** [comment]
I might be wrong, but the errors in ansible/check look more like configuration issues in the testing pipeline.
Please advise, if you think I can do something from my side.

**Comment by cvstebut** [comment]
recheck

**Comment by gravesm** [comment]
@cvstebut The sanity failure for the 2.11 test is unrelated to your changes here, and will likely have to be fixed in ansible. Not sure yet what the time frame for that fix is.

**Comment by cvstebut** [comment]
@gravesm Thanks for letting me know! And also many thanks to those working on the testing pipelines!

**Comment by cvstebut** [comment]
Hi,

I am currently working on adding further optional parameters to helm_template: 

- name (str: sets helm template NAME, thus setting the release name to be used in the rendered templates)
- release_name (bool: sets helm template --release-name parameter. The release_name will then be used in the output_dir path)
- disable_hook (bool: sets helm template --no_hooks parameter)

Should I extend this PR or wait for this PR to be merged and start the next one?
Waiting would also allow to "step back" while all the work on the integration tests is being done.

**Review by gravesm — APPROVED** [review]

**Comment by goneri** [comment]
The ansible-ee-tests-stable-2.12 has failed. I just pushed https://github.com/ansible/ansible-zuul-jobs/pull/1369 to ignore it. Once this PR is merged, we will able to retrigger another gating job with the `regate` key word.

**Comment by goneri** [comment]
regate

**Comment by goneri** [comment]
My previous PR didn't work. I've pushed this one https://github.com/ansible/ansible-zuul-jobs/pull/1370 which this time turn the job non-voting globally.

**Comment by goneri** [comment]
regate

**Pull request merged**
